### PR TITLE
feat(orchestration): Allow sibling stages to continue on FAILED_CONTINUE

### PR DESCRIPTION
**Pull request by marchello2000**
Normally, if a synthetic child stage fails with `FAILED_CONTINUE` the `CompleteStageHandler`
will propagate the `FAILED_CONTINUE` status to the parent. This will prevent all subsequent
sibling stages from executing.
This change allows for an option (similar to Tasks) to continue execution even when a
child stage has a `status=FAILED_CONTINUE`

This option can only be set on a stage that is a synthetic child.
This is needed for Monitored Deploy where we want to allow the deploy to proceed even if the
monitoring fails

### CODE REVIEW

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt [hunk 1]
```diff
@@ -142,7 +142,7 @@ class CompleteStageHandler(
 
         // When a synthetic stage ends with FAILED_CONTINUE, propagate that status up to the stage's
         // parent so that no more of the parent's synthetic children will run.
-        if (stage.status == FAILED_CONTINUE && stage.syntheticStageOwner != null) {
+        if (stage.status == FAILED_CONTINUE && stage.syntheticStageOwner != null && stage.allowSiblingStagesToContinueOnFailure) {
```
Comment (ajordens) [inline orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: Shouldn't this be `!stage.allow...`? (based on the comment of not wanting to propagate up to the parent if siblings are allowed to run)

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java [hunk 2]
```diff
@@ -64,6 +64,16 @@ public Object get(@Nullable Object key) {
...
   }
 
+  /**
+   * Get a value from the current context ONLY - never looking at the ancestors' outputs
+   *
+   * @param key The key to look
+   * @param defaultValue default value to return if key is not present
+   * @return value or null if not present
+   */
+  Object getCurrentOnly(@Nullable Object key, Object defaultValue) {
```
Comment (ajordens) [inline orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: Not sure how I feel about `getCurrentOnly` vs overloading `get()` -- will let someone else weigh in.

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt (hunk 1 above)
Comment (marchello2000) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: Yes, thank you! change the name of the var last minute that inverted the meaning and forgot to update this. i guess i should find and update tests for this

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java (hunk 2 above)
Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: you mean something like 
```
public Object get(@Nullable Object key, Object defaultValue, boolean localOnly)
```
?

File: orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt (hunk 1 above)
Comment (marchello2000) [reply orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt]: oh, look, the existing tests caught it... fixed up and added a new test

File: orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java (hunk 2 above)
Comment (jeyrschabu) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: +1 on overloading `get(...)`

Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: Just curious, why?

I think
```
stage.context.getCurrentOnly("someValue", false);
```

is more descriptive than:
```
stage.context.get("someValue", false, true);
```

Comment (jeyrschabu) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: get() because the context is a map. Matter of preference.

Comment (marchello2000) [reply orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/model/StageContext.java]: I hear you. This area has caused some confusion in the past (where something comes from in the context) so I am more inclined to make this very explicit

**Comment by marchello2000** [comment]
Going to merge this because there are teams blocked on this. If you have feedback please give it to me and I will incorporate

**Pull request merged**